doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+
+The simplest case to analyze is when the kernel just slides across every
+position of the input (i.e., :math:`s = 1` and :math:`p = 0`).
+Here is an example for :math:`i = 4` and :math:`k = 3`:
+
+.. figure:: conv_arithmetic_figures/no_padding_no_strides.gif
+
+One way of defining the output size in this case is by the number of possible
+placements of the kernel on the input. Let's consider the width axis: the kernel
+starts on the leftmost part of the input feature map and slides by steps of one
+until it touches the right side of the input. The size of the output will be
+equal to the number of steps made, plus one, accounting for the initial position
+of the kernel. The same logic applies for the height axis.
+
+More formally, the following relationship can be inferred:
+
+.. admonition:: Relationship 1
+
+    .. code-block:: python
+
+        output = theano.tensor.nnet.conv2d(
+            input, filters, input_shape=(i, i), filter_shape=(c1, c2, k, k),
+            border_mode=(0, 0), subsample=(1, 1))
+        # output.shape[2] == output.shape[3] == (i - k) + 1
+
+Zero padding, unit strides
+--------------------------
+
+To factor in zero padding (i.e., only restricting to :math:`s = 1`), let's
+consider its effect on the effective input size: padding with :math:`p` zeros
+changes the effective input size from :math:`i` to :math:`i + 2p`. In the
+general case, Relationship 1 can then be used to infer the following
+relationship:
+
+.. admonition:: Relationship 2
+
+    For any :math:`i`, :math:`k` and :math:`p`, and for :math:`s = 1`,
+
+    .. code-block:: python
+
+        output = theano.tensor.nnet.conv2d(
+            input, filters, input_shape=(i, i), filter_shape=(c1, c2, k, k),
+            border_mode=(p, p), subsample=(1, 1))
+        # output.shape[2] == output.shape[3] == (i - k) + 2 * p + 1
+
+Here is an example for :math:`i = 5`, :math:`k = 4` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/arbitrary_padding_no_strides.gif
+
+Special cases
+-------------
+
+In practice, two specific instances of zero padding are used quite extensively
+because of their respective properties. Let's discuss them in more detail.
+
+Half (same) padding
+^^^^^^^^^^^^^^^^^^^
+
+Having the output size be the same as the input size (i.e., :math:`o = i`) can
+be a desirable property:
+
+.. admonition:: Relationship 3
+
+    For any :math:`i` and for :math:`k` odd (:math:`k = 2n + 1, \quad n \in
+    \mathbb{N}`), :math:`s = 1` and :math:`p = \lfloor k / 2 \rfloor = n`,
+
+    .. code-block:: python
+
+        output = theano.tensor.nnet.conv2d(
+            input, filters, input_shape=(i, i), filter_shape=(c1, c2, k, k),
+            border_mode='half', subsample=(1, 1))
+        # output.shape[2] == output.shape[3] == i
+
+This is sometimes referred to as *half* (or *same*) padding. Here is an example
+for :math:`i = 5`, :math:`k = 3` and (therefore) :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/same_padding_no_strides.gif
+
```

Comment:
I would also add a note to specify that some frameworks allow even kernels and/or might implement the `same` convolution differently (e.g. in keras it's `(i + s - 1) // s`)